tableau-databricks/dialect.tdd

```diff
@@ -116,15 +116,187 @@ limitations under the License.
+    </function>
+    <function group='string' name='STARTSWITH' return-type='bool'>
+      <formula>&#10;STARTSWITH(%1, %2)</formula>
+      <argument type='str' />
+      <argument type='str' />
+    </function>
+     <function group='string' name='ENDSWITH' return-type='bool'>
+      <formula>&#10;ENDSWITH(%1, %2)</formula>
+      <argument type='str' />
+      <argument type='str' />
+    </function>
+
+    <!-- tableau returns 1/0 instead of true/false>
+     <function group='cast' name='STR' return-type='str'>
+      <formula>CAST(%1 AS STRING)</formula>
+      <argument type='bool' />
+    </function>
+    <-->
+
+    <function group='cast' name='INT' return-type='int'>
+      <formula>CAST(%1 AS BIGINT)</formula>
+      <argument type='bool' />
+    </function>
+
+    <function group='cast' name='FLOAT' return-type='real'>
+      <formula>CAST(%1 AS DOUBLE)</formula>
+      <argument type='bool' />
+    </function>
+
+    <function group='aggregate' name='COUNT' return-type='int'>
+      <formula>COUNT(%1)</formula>
+      <unagg-formula>(CASE WHEN (%1 IS NULL) THEN 0 ELSE 1 END)</unagg-formula>
+      <argument type='bool' />
+    </function>
+    <function group='aggregate' name='COUNTD' return-type='int'>
+      <formula>COUNT(DISTINCT %1)</formula>
+      <unagg-formula>(CASE WHEN (%1 IS NULL) THEN 0 ELSE 1 END)</unagg-formula>
+      <argument type='bool' />
+    </function>
+
+    <function group='numeric' name='COT' return-type='real'>
+      <formula>IF(%1 != 0, COT(%1), NULL)</formula>
+      <argument type='real' />
+    </function>
+
+    <function group='numeric' name='LN' return-type='real'>
+      <formula>LN(%1)</formula>
+      <argument type='real' />
+    </function>
+
+    <function group='numeric' name='LOG' return-type='real'>
+      <formula>LOG10(%1)</formula>
+      <argument type='real' />
+    </function>
+    <function group='numeric' name='LOG' return-type='real'>
+      <formula>LOG(%2, %1)</formula>
+      <argument type='real' />
+      <argument type='real' />
+    </function>
+
+     <function group='numeric' name='TAN' return-type='real'>
+      <formula>TAN(%1)</formula>
+      <argument type='real' />
+    </function>
+
+
     <date-function name='DATENAME' return-type='str'>
-        <formula>
-            FORMAT_NUMBER(FLOOR((14 + DATEDIFF(%2, TRUNC(%2,'YY')) + DATEDIFF(TRUNC(%2,'YY'), NEXT_DAY(TRUNC(%2,'YY'), 'MO')))/7), 0)
-        </formula>
         <formula part='quarter'>CAST(QUARTER(%2) AS STRING)</formula>
         <formula part='year'>CAST(YEAR(%2) AS STRING)</formula>
         <formula part='month'>CAST(MONTH(%2) AS STRING)</formula>
         <formula part='week'>
-            FORMAT_NUMBER(FLOOR((14 + DATEDIFF(%2, TRUNC(%2,'YY')) + DATEDIFF(TRUNC(%2,'YY'), NEXT_DAY(TRUNC(%2,'YY'), 'SU')))/7), 0)
+            FORMAT_NUMBER(FLOOR((14 + DATEDIFF(%2, TRUNC(%2,&apos;YY&apos;)) + DATEDIFF(TRUNC(%2,&apos;YY&apos;), NEXT_DAY(TRUNC(%2,&apos;YY&apos;), &apos;SU&apos;)))/7), 0)
```

Comment:
Nit: Any reason we are using `&apos`?

Reply:
I'm just trying to make the format consistent, i.e. always use the html entity name. That seems the format other tableau dialect is following: https://github.com/tableau/connector-plugin-sdk/blob/master/samples/components/dialects/Hive12Dialect.tdd